pegasus/helpers/section_api_helpers.rb

```diff
@@ -252,6 +252,14 @@ def self.valid_courses(user_id = nil)
         end
   end
 
+  def self.progress_disabled_courses(user_id = nil)
+    disabled_courses = valid_courses(user_id).select do |course|
+      script_name = Script.find(course[:id]).name
```

Comment:
Any reason not to use the script cache (via `Script.get_from_cache(course[:id]).name`)?